[Merged by Bors] - chore: add githelper.py script

Garmelon opened the pull request:
As described in the readme file, this somewhat opinionated script aims to help fix weird git repo setups and restore them to a standardized state that closely matches what `gh repo clone` does. Like `migrate_to_fork.py`, this script requires the user to have `git` and `gh` installed.

To use the script, run `scripts/githelper.py fix` in a git repo that is either a clone of mathlib or of a mathlib fork. The script will prompt you before making any changes, so the user retains control over the entire process. The script also prints the commands used.

`fix` is a subcommand so other git and github management tasks can be added in the future. Possible examples: Deleting branches of closed PRs, keeping the fork in sync with upstream, removing branches from a fork that was accidentally cloned with all mathlib branches included.

---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

If you are moving or deleting declarations, please include these lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

Garmelon committed:
Fix remote 'upstream' and gh default repo

Garmelon committed:
Locate mathlib fork (even if renamed)

Garmelon committed:
Fix remote 'origin', fix remotes of branch 'master'

Forks if necessary

Garmelon committed:
Default to yes for prompts

Garmelon committed:
Remove unnecessary comment

Garmelon committed:
Offer to rename remotes instead of always recreating

Garmelon committed:
Fix some bugs

Garmelon committed:
Describe script in README

Garmelon committed:
Fix colors on Windows

jcommelin commented on scripts/README.md:
```diff
@@ -55,6 +55,18 @@ to learn about it as well!
+  The target state mostly matches the state after of a freshly cloned fork (`gh repo clone <fork>`)
+  and looks like this:
+
+  - The branch `upstream` points to `leanprover-community/mathlib4`
+  - The branch `origin` points to the contributor's own fork
```
```suggestion
  - The remote `upstream` points to `leanprover-community/mathlib4`
  - The remote `origin` points to the contributor's own fork
```

Garmelon replied:
Oops

Garmelon committed:
Fix config editing commands

git config get and git config set were only added in 2.46.0, but some
platforms (e.g. macOS) ship older git versions.

Garmelon committed:
Fix typos in readme

bryangingechen commented:
Feel free to ping me after you've made the changes you want to! As I said on Zulip, I think the guide on the website can still be improved, so feel free to open a PR there as well if you have additional suggestions.

Garmelon commented:
After having another look at the git guide, my script performs steps 1 to 4 and only then deviates: Instead of step 5, it configures `git config branch.master.pushRemote origin`. 

I personally prefer this since it is closer to the default git setup. Every nonstandard option has the possibility of leading to unexpected, hard-to-google issues.

If this deviation is fine, feel free to merge this PR, @bryangingechen. I have no further changes planned for now.
